### PR TITLE
Dashboard Overview

### LINKED ISSUE #71: Overview dashboard
Implement overview dashboard for picking what diagram to display.

### PR DESCRIPTION
**Pull request by Caw3**
Fixes #71
Old upload page for setting to localstorage is still reachable at /test/upload.

### CODE REVIEW

File: src/components/dashboard/dividerStack.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import { styled } from '@mui/material/styles'
+
+const Item = styled(Paper)(({ theme }) => ({
+  backgroundColor: theme.palette.mode === 'dark' ? '#1A2027' : '#fff',
+  ...theme.typography.body2,
+  padding: theme.spacing(1),
+  textAlign: 'center',
+  marginTop: '20px',
+  color: theme.palette.text.secondary
+}))
```
Comment (JoXW100) [inline src/components/dashboard/dividerStack.js:15]: should be moved to a separate file, or into the function body

File: src/components/dashboard/dividerStack.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+export default function DividerStack () {
```
Comment (JoXW100) [inline src/components/dashboard/dividerStack.js:17]: be consistent and use other function declaration method

File: src/components/dashboard/dividerStack.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import * as React from 'react'
+import Divider from '@mui/material/Divider'
+import Typography from '@mui/material/Typography'
+import Paper from '@mui/material/Paper'
+import Stack from '@mui/material/Stack'
+import { styled } from '@mui/material/styles'
```
Comment (JoXW100) [inline src/components/dashboard/dividerStack.js:6]: No need to import react like this
```suggestion
import Divider from '@mui/material/Divider'
import Typography from '@mui/material/Typography'
import Paper from '@mui/material/Paper'
import Stack from '@mui/material/Stack'
import { styled } from '@mui/material/styles'
```

File: src/components/dashboard/uploadButton.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import * as React from 'react'
+import IconButton from '@mui/material/IconButton'
+import AddCircleOutlinedIcon from '@mui/icons-material/AddCircleOutlined'
+import styles from '../../styles/dashboardStyles.module.css'
```
Comment (JoXW100) [inline src/components/dashboard/uploadButton.js:4]: No need to import react like this
```suggestion
import IconButton from '@mui/material/IconButton'
import AddCircleOutlinedIcon from '@mui/icons-material/AddCircleOutlined'
import styles from '../../styles/dashboardStyles.module.css'
```

File: src/styles/dashboardStyles.module.css [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+}
+.uploadButtonText{
+	margin-left:1rem;
+    color: #e4e4e4;
+	line-height: 1.6;
+	font-weight: 500;
+	font-size: 1.25rem;
+	font-family: "Roboto","Helvetica","Arial",sans-serif;
+ }
+
```
Comment (JoXW100) [inline src/styles/dashboardStyles.module.css:62]: Strange indentation in this file

File: src/utils/localStorage.js [hunk 6]
```diff
@@ -16,6 +17,7 @@ function getDiagram () {
  * @param diagram { GraphMap } - An object that describes the diagram
  */
 function setDiagram (diagram) {
+  console.log(diagram)
   window.localStorage.setItem('diagram', JSON.stringify(diagram))
 }
```
Comment (JoXW100) [inline src/utils/localStorage.js:23]: Remove debug functions in this file, (all `console.log`)

File: test/database.test.js [hunk 7]
```diff
@@ -58,12 +58,12 @@ describe('getAll', () => {
...
   it('calls find without a query', async () => {
     await db.getAll(col)
-    expect(conn.find).toHaveBeenCalledWith({})
+    expect(conn.find).toHaveBeenCalledWith({},{})
   })
   it('calls find with a query', async () => {
     const testQuery = { _id: '123' }
     await db.getAll(col, testQuery)
-    expect(conn.find).toHaveBeenCalledWith(testQuery)
+    expect(conn.find).toHaveBeenCalledWith(testQuery,{})
```
Comment (JoXW100) [inline test/database.test.js:66]: consider using a default value in this function

**Review by JoXW100 — CHANGES_REQUESTED** [review]

File: test/database.test.js (hunk 7 above)
Comment (Caw3) [reply test/database.test.js:66]: I do not understand. The function needs a specific value?

File: src/components/dashboard/dashSearchField.js [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+  '&:hover': {
+    backgroundColor: alpha(theme.palette.common.white, 0.25)
+  },
+  marginLeft: 0,
+  width: '100%',
+  [theme.breakpoints.up('sm')]: {
+    marginLeft: theme.spacing(1),
+    width: 'auto'
+  }
+}))
```
[the remark this answers is not in the document]
Comment (Caw3) [reply src/components/dashboard/dashSearchField.js:19]: This should be refactored later together with the identical components in navigationBar.js

**Review by JoXW100 — APPROVED** [review]

File: src/components/dashboard/dashAlert.js [hunk 9]
```diff
@@ -0,0 +1,35 @@
+import Alert from '@mui/material/Alert'
+import AlertTitle from '@mui/material/AlertTitle'
+import IconButton from '@mui/material/IconButton'
+import Collapse from '@mui/material/Collapse'
+import CloseIcon from '@mui/icons-material/Close'
+
+const DashAlert = ({ severity, alertOpen, setAlertOpen }) => {
+  return (
+    <Collapse in={!!alertOpen}>
```
Comment (thomssonkarl) [inline src/components/dashboard/dashAlert.js:9]: so just alterOpen? (!!)

File: src/components/dashboard/dashAlert.js [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+              setAlertOpen(null)
+            }}
+          >
+            <CloseIcon fontSize='inherit' />
+          </IconButton>
+          }
+        sx={{ mb: 2, borderRadius: '0' }}
+      >
+        <AlertTitle>
+          {alertOpen}
```
Comment (thomssonkarl) [inline src/components/dashboard/dashAlert.js:28]: 'alertOpen' is quite confusing, usually we use xOpen for open/close states but this seems like it is some text.

File: src/components/dashboard/dividerStack.js (hunk 2 above)
Comment (thomssonkarl) [reply src/components/dashboard/dividerStack.js:16]: Agreed keep it the same as the rest

File: src/components/dashboard/loginGroup.js [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+      <>
+        <div onClick={handleMenu} className={styles.loginMenu}>
+          <div className={styles.userText}>
+            {user.nickname}
+          </div>
+          <ManageAccountsIcon />
+        </div>
+
+        <Menu
+          id='menu-appbar'
```
Comment (thomssonkarl) [inline src/components/dashboard/loginGroup.js:38]: Id can be removed here if not used

File: src/pages/index.js [hunk 12]
```diff
@@ -1,72 +1,83 @@
...
-        processLeader: 'xxx yyy',
-        lastApproved: '2022-05-02',
-        fileLoaded: false
-      }
-    }
-    setState(files)
-    setMetadata(meta)
-  }, [])
+    checkSession().then(() => {
+      console.log('session was checked')
```
Comment (thomssonkarl) [inline src/pages/index.js:37]: This should probably not be in production? (console.log)

**Review by thomssonkarl — APPROVED** [review]

**Pull request merged**